src/Microsoft.AspNet.Mvc.Core/Rendering/Html/HtmlHelper.cs

```diff
@@ -961,9 +962,8 @@ protected virtual string GenerateValue(string expression, object value, string f
             {
                 if (string.IsNullOrEmpty(expression))
                 {
-                    // case 2(a): format the value from ModelMetadata for the current model
-                    var metadata = ViewData.ModelMetadata;
-                    resolvedValue = FormatValue(metadata.Model, format);
+                    // case 2(a): format the value from ViewData for the current model
+                    resolvedValue = FormatValue(ViewData.Model, format);
```

Comment:
so, have you changed things enough that `ViewData.Model == ModelMetadata.Model` is an invariant?